build/acm.mk

```diff
@@ -0,0 +1,205 @@
+# ACM (Advanced Cluster Management) installation targets for OpenShift
+# This file is specific to downstream OpenShift/OCP features only
+
+.PHONY: acm-install acm-mce-install acm-operator-install acm-instance-install acm-status acm-import-cluster acm-uninstall
+
+##@ ACM (OpenShift only)
+
+acm-install: acm-mce-install acm-operator-install acm-instance-install ## Install MCE, ACM operator and instance
+
+# Install MultiCluster Engine (required for ACM)
+acm-mce-install:
+	@echo "Installing MultiCluster Engine (MCE)..."
+	@echo "Creating multicluster-engine namespace..."
+	@oc create namespace multicluster-engine --dry-run=client -o yaml | oc apply -f -
+	@echo "Creating MCE OperatorGroup..."
+	@printf '%s\n' \
+		'apiVersion: operators.coreos.com/v1' \
+		'kind: OperatorGroup' \
+		'metadata:' \
+		'  name: multicluster-engine' \
+		'  namespace: multicluster-engine' \
+		'spec:' \
+		'  targetNamespaces:' \
+		'  - multicluster-engine' \
+		| oc apply -f -
+	@echo "Creating MCE Subscription..."
+	@printf '%s\n' \
+		'apiVersion: operators.coreos.com/v1alpha1' \
+		'kind: Subscription' \
+		'metadata:' \
+		'  name: multicluster-engine' \
+		'  namespace: multicluster-engine' \
+		'spec:' \
+		'  channel: stable-2.9' \
+		'  name: multicluster-engine' \
+		'  source: redhat-operators' \
+		'  sourceNamespace: openshift-marketplace' \
+		| oc apply -f -
+	@echo "Waiting for MCE operator CSV to be ready..."
+	@for i in $$(seq 1 60); do \
+		if oc get csv -n multicluster-engine -o name 2>/dev/null | grep -q multicluster-engine; then \
+			echo "MCE CSV found, waiting for Succeeded phase..."; \
+			oc wait --for=jsonpath='{.status.phase}'=Succeeded csv -l operators.coreos.com/multicluster-engine.multicluster-engine -n multicluster-engine --timeout=300s && break; \
+		fi; \
+		echo "  Waiting for MCE CSV to appear ($$i/60)..."; \
+		sleep 5; \
+	done
+	@echo "Creating MultiClusterEngine instance..."
+	@printf '%s\n' \
+		'apiVersion: multicluster.openshift.io/v1' \
+		'kind: MultiClusterEngine' \
+		'metadata:' \
+		'  name: multiclusterengine' \
+		'spec: {}' \
+		| oc apply -f -
+	@echo "Waiting for ManagedCluster CRD to be available..."
+	@for i in $$(seq 1 120); do \
+		if oc get crd managedclusters.cluster.open-cluster-management.io >/dev/null 2>&1; then \
+			echo "✅ ManagedCluster CRD is now available!"; \
+			break; \
+		fi; \
+		echo "  Waiting for ManagedCluster CRD ($$i/120)..."; \
+		sleep 5; \
+	done
+	@echo "✓ MCE installation complete"
+
+# Install ACM operator (Subscription, OperatorGroup, etc.)
+acm-operator-install:
```

Comment:
I feel like these would be better as scripts that are called by the make targets. It'd make the Makefile much cleaner.